app/controllers/teacher_interface/uploads_controller.rb

```diff
@@ -25,15 +25,23 @@ def create
     end
 
     def delete
+      @delete_file_form = DeleteFileForm.new
     end
 
     def destroy
-      if ActiveModel::Type::Boolean.new.cast(params.dig(:upload, :confirm))
-        @upload.attachment.purge
-        @upload.destroy!
+      confirm = params.dig(:teacher_interface_delete_file_form, :confirm)
+      @delete_file_form = TeacherInterface::DeleteFileForm.new(confirm:)
+      if @delete_file_form.valid?
+        if @delete_file_form.confirm
+          @upload.attachment.purge
+          @upload.destroy!
```

Comment:
I wonder if it's worth moving this logic in to the form, so it can easily be unit tested.

Similar to the forms we've got with a `def save` method which updates a model, we could have a method which does the same:

```rb
def save
  return unless valid?

  upload.attachment.purge
  upload.destroy!
end
```

The controller becomes much simpler then, we would need to pass the `upload` in to the form in the same way that we do with `application_form` in other forms.